src/main/kotlin/io/fabric8/launcher/booster/Files.kt

```diff
@@ -0,0 +1,121 @@
+ */
+
+package io.fabric8.launcher.booster
+
+import java.io.File
+import java.io.IOException
+import java.io.OutputStream
+import java.nio.file.FileVisitResult
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.SimpleFileVisitor
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.function.Predicate
+import java.util.zip.ZipEntry
+import java.util.zip.ZipFile
+import java.util.zip.ZipInputStream
+import java.util.zip.ZipOutputStream
+
+/**
+ * @author [George Gastaldi](mailto:ggastald@redhat.com)
+ */
+object Files {
+    fun removeFileExtension(file: String): String {
+        val idx = file.lastIndexOf('.')
+        return if (idx > 0) file.substring(0, idx) else file
+    }
+
+    /**
+     * Zips an entire directory and stores in the provided [OutputStream]
+     *
+     * @param root      the root directory to be used
+     * @param directory the directory to be zipped
+     * @param os        the [OutputStream] which the zip operation will be written to
+     * @throws IOException if any I/O error happens
+     */
+    @Throws(IOException::class)
+    fun zip(root: String, directory: Path, os: OutputStream, filter: Predicate<Path>) {
+        ZipOutputStream(os).use { zos ->
+            java.nio.file.Files.walkFileTree(directory, object : SimpleFileVisitor<Path>() {
+                @Throws(IOException::class)
+                override fun visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult {
+                    if (!filter.test(file)) {
+                        return FileVisitResult.CONTINUE
+                    }
+                    val entry = root + File.separator + directory.relativize(file).toString()
+                    zos.putNextEntry(ZipEntry(entry))
+                    java.nio.file.Files.copy(file, zos)
+                    zos.closeEntry()
+                    return FileVisitResult.CONTINUE
+                }
+
+                @Throws(IOException::class)
+                override fun preVisitDirectory(dir: Path, attrs: BasicFileAttributes): FileVisitResult {
+                    if (!filter.test(dir)) {
+                        return FileVisitResult.SKIP_SUBTREE
+                    }
+                    val entry = root + File.separator + directory.relativize(dir).toString() + File.separator
+                    zos.putNextEntry(ZipEntry(entry))
+                    zos.closeEntry()
+                    return FileVisitResult.CONTINUE
+                }
+            })
+        }
+    }
+
+    /**
+     * Unzips a ZIP file in the target directory, preserving the directory structure
+     *
+     * @param zipFile
+     * @param targetDir
+     * @throws IOException
+     */
+    @Throws(IOException::class)
+    fun unzip(zipFile: Path, targetDir: Path) {
+        ZipFile(zipFile.toFile()).use { zf ->
+            zf.stream().forEach( { ze ->
+                val newEntry = Paths.get(targetDir.toString(), ze.name)
+                if (ze.isDirectory) {
```

Comment:
Don't forget to add this verification: https://github.com/fabric8-launcher/launcher-backend/blob/master/base/src/main/java/io/fabric8/launcher/base/Paths.java#L54-L57



More info: https://snyk.io/research/zip-slip-vulnerability